service/discovery/azure/arm_template.go

```diff
@@ -27,22 +27,24 @@ package azure
 
 import (
 	"context"
+	"errors"
 	"fmt"
-
 	"strings"
 
 	"clouditor.io/clouditor/api/discovery"
 	"clouditor.io/clouditor/voc"
 	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2016-02-01/resources"
+	//TODO(garuppel): The bootDiagnosticsStorageURI is gone? Where can I find it in the ARM template. Do we need to update the API to '2020-10-01/resources'?
+	//"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2020-10-01/resources"
 	"github.com/Azure/go-autorest/autorest/to"
 )
 
-type azureIacTemplateDiscovery struct {
+type azureArmTemplateDiscovery struct {
```

Comment:
Since we are in the `azure` package, you could argue that azure in `azureXXX` struct names is redundant as well. But on the other side, `ARMTemplateDiscovery` would be exported due to the starting capital letter. If, e.g., `templateDiscovery` only  doesn't fit (which then could be consistently used in discoveries of other services if they provide templates), I also think it is fine this way.